qiskit_experiments/library/stark/ramsey_xy.py

```diff
@@ -0,0 +1,319 @@
+
+            (Ramsey Y) The pulse before measurement rotates by pi-half around the Y axis
+
+                     ┌────┐┌────────┐┌───┐┌───────────────┐┌───────────┐┌────┐┌─┐
+                  q: ┤ √X ├┤ StarkV ├┤ X ├┤ StarkU(delay) ├┤ Rz(-3π/2) ├┤ √X ├┤M├
+                     └────┘└────────┘└───┘└───────────────┘└───────────┘└────┘└╥┘
+                c: 1/══════════════════════════════════════════════════════════╩═
+                                                                               0
+
+        In principle, the sequence is a variant of :class:`RamseyXY` circuit,
+        however, the delay in between √X gates is replaced with the off-resonant drive.
+        This off-resonant drive causes the Stark shift on the qubit that modulates
+        qubit frequency under the drive, which accumulates phase during the
+        Ramsey sequence against the shifted frequency.
+        This frequency shift is a function of the offset of the Stark tone from the
+        qubit frequency, and magnitude of the tone.
+
+        Note that the Stark tone takes the form of the flat-topped Gaussian envelope,
+        in which the magnitude of pulse varies in time in rising and falling edges,
+        resulting in the time-dependent Stark shift.
+        This is hardly characterized by the Ramsey experiment, and thus an additional pulse
+        involving only edges (StarkV) is added in front of the primary Stark drive (StarkU).
+        The sign of phase accumulation is inverted with the X gate in the middle to
+        cancel out the unwanted phase accumulation in the pulse edges.
+
+        This technique allows the experiment to only accumulate the net phase
+        during the flat-top part of the StarkU pulse with the constant magnitude.
+
+    # section: note
+
+        In the lower power limit, the Stark shift has a quadratic dependency on the
+        pulse amplitude. This implies the resulting Stark shift is not sensitive to
+        the sign of the amplitude of StarkV and StarkU pulses on one hand.
+        On the other hand, the sign of the shift depends on the sign of frequency offset.
+        In other words, positive (negative) Stark shift occurs when the Stark tone frequency
+        is lower (higher) than the qubit frequency.
+
+        In this experiment, the amplitude of Stark tone, i.e. `stark_amp` parameter, is
+        abstracted to give positive (positive) Stark shift with positive (negative) stark_amp
+        for better handling of the experiment parameters.
+        The StarkV and StarkU pulses are always positive regardless of the sign of stark_amp,
+        while the sign of frequency offset is implicitly determined by its sign.
+
+    # section: analysis_ref
+        :py:class:`RamseyXYAnalysis`
+
+    # section: see_also
+        qiskit_experiments.library.characterization.ramsey_xy.RamseyXY
+
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        stark_amp: float,
```

Comment:
I wonder if `stark_amp` is really more special than the other options and needs to be split out as a required parameter?

Reply:
Yes, I think so. This experiment intends to measure RamseyXY at a particular qubit frequency that is determined by the stark_amp. If we provide random default value, user will never know what they are measuring.